### ISSUE TITLE
Add support for cursor query parameter

### ISSUE THREAD

**Issue by fmrsabino** [5afe/safe-client-gateway-nest#62]
- Current paginated endpoints should have a `cursor` parameter (as implemented in the current version of the Safe Client Gateway).
- We should add support for it while mapping the contents of the cursor to the correct pagination query parameters of the Safe Transaction Service and the Safe Config Service.

Reference implementation:
- https://github.com/safe-global/safe-client-gateway/blob/main/src/common/models/page.rs
- https://github.com/safe-global/safe-client-gateway/blob/main/src/common/converters/page_metadata.rs

**Comment by moisses89** [comment]
I didn't like a lot the cursor parameter, but I understand that it was implemented on client-gateway then should implemented also here to avoid changes from clients. 
Are we going to eliminate this cursor in the future? :)

**Comment by fmrsabino** [comment]
> I didn't like a lot the cursor parameter, but I understand that it was implemented on client-gateway then should be implemented also here to avoid changes from clients. Are we going to eliminate this cursor in the future? :)

Yes for the current stage we should match the pagination as-is from the Safe Client Gateway. However in later stages we can always review the best format moving forward. There needs to be some pagination in place in the end but we can see which one fits our needs the best.
